### PR TITLE
Adding files for Managed Instances, Managed Databases, and Instance Failover Groups

### PR DESCRIPTION
**Pull request by jugeorge**
## Description
These are the generated files (and corresponding tests) for Managed Instances, Managed Databases, and Instance Failover Groups. These were merged in with the following PRs:

https://github.com/Azure/azure-rest-api-specs/pull/2706
https://github.com/Azure/azure-rest-api-specs/pull/2615
https://github.com/Azure/azure-rest-api-specs/pull/2892
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs [hunk 1]
```diff
@@ -31,7 +31,10 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "FailoverGroups", "2015-05-01-preview"),
                 new Tuple<string, string, string>("Sql", "FirewallRules", "2014-04-01"),
                 new Tuple<string, string, string>("Sql", "GeoBackupPolicies", "2014-04-01"),
+                new Tuple<string, string, string>("Sql", "InstanceFailoverGroups", "2017-10-01-preview"),
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: Please run `msbuild build.proj /t:build /p:Scope=SDKs\SqlManagement` and commit any artifacts that may be generated

File: src/SDKs/SqlManagement/Sql.Tests/Utilities/SqlManagementTestContext.cs [hunk 2]
```diff
@@ -92,6 +92,35 @@ public Server CreateServer(ResourceGroup resourceGroup, string location)
 
             return v12Server;
         }
+		
+		public ManagedInstance CreateManagedInstance(ResourceGroup resourceGroup)
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Sql.Tests/Utilities/SqlManagementTestContext.cs]: nit: Please fix indentation
in other places too

File: src/SDKs/_metadata/sql_resource-manager.txt [hunk 3]
```diff
@@ -1,14 +1,11 @@
-
-Type dsc to start/install the DS Consolidated Console
-
-2018-04-06 18:59:06 UTC
+2018-04-18 15:16:26 UTC
```
Comment (dsgouda) [inline src/SDKs/_metadata/sql_resource-manager.txt]: Please create a `generate.ps1` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Needs a few minor changes.

File: src/SDKs/_metadata/sql_resource-manager.txt (hunk 3 above)
Comment (jugeorge) [reply src/SDKs/_metadata/sql_resource-manager.txt]: @dsgouda when I try and run the generate.ps1 file I've added, I get "Could not find module: C:\azure-sdk-for-net\tools\SdkBuildTools\psModules\CodeGenerationModules\generateDotNetSdkCode.psm1. Please run msbuild build.proj to install the module." However, this file is in the folder and the msbuild is passing. Any ideas why?

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs (hunk 1 above)
Comment (jugeorge) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: no new files were generated

Comment (dsgouda) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: This file should have some changes
https://github.com/jugeorge/azure-sdk-for-net/blob/8862cab86643c1c7dadf6228ceba37c277d87332/src/SDKs/SqlManagement/AzSdk.RP.props

**Review by dsgouda — APPROVED** [review]
LGTM subject to builds passing.

**Comment by dsgouda** [comment]
Overriding CI since last commit should not have any effect

**Pull request merged**